applications/dashboard/models/class.userauthenticationnoncemodel.php

```diff
@@ -31,7 +41,85 @@ public function insert($fields) {
         if (!isset($fields['Timestamp'])) {
             $fields['Timestamp'] = date(MYSQL_DATE_FORMAT);
         }
+        $result = parent::insert($fields);
+
+        return $result;
+    }
+
+    /**
+     * Issue a signed Nonce.
+     *
+     * @param string $expires The expiration time of the nonce.
+     * @param string $type The type of nonce.
+     * @return string $nonce The signed nonce.
+     * @throws Gdn_UserException Unable to generate a signed nonce.
+     */
+    public function issue($expires = '5 minutes', $type = 'system'): string {
+        $token = $this->randomToken();
+        $nonce = $this->signToken($token, $expires);
+        $expireDate = Gdn_Format::toDateTime($this->toTimestamp($expires));
+
+        $result = $this->insert([
+            'Nonce' => $nonce,
+            'Token' => $type,
+            'Timestamp' => $expireDate,
+        ]);
+
+        if ($result === false) {
+            throw new Gdn_UserException($this->Validation->resultsText(), 400);
+        }
+        return $nonce;
+    }
+
+    /**
+     * Consumes the nonce and invalidates the timestamp so it can't be used again.
+     *
+     * @param string $nonce The nonce to be consumed.
+     * @throws Exception if unable to find nonce.
+     */
+    public function consume(string $nonce) {
+        $row = $this->getID($nonce, DATASET_TYPE_ARRAY);
+        if ($row) {
+            // Timestamp cannot be null or zero. Use a constant date for consumed nonces.
+            parent::update(
```

Comment:
What would be the benefit of calling the parent's `update` method, instead of the inherited method (e.g. `$this->update()`)?